documentation/docs/30-advanced/40-service-workers.md

```diff
@@ -4,10 +4,75 @@ title: Service workers
+
+The following example caches all files from the build eagerly and also caches all other requests occuring later on. This would make each page work offline once visited.
+
+```js
+// @ts-nocheck Official TS Service Worker typings are still a work in progress.
+import { build, files, version } from '$service-worker';
+
+// Give the current service worker instance a unique cache name
+const CACHE_NAME = `cache-${version}`;
+// In this simple example, we just cache all files
+const to_cache = build.concat(files);
+
+self.addEventListener('install', (event) => {
+	// Create new cache and add all files to it
+	async function addFilesToCache() {
+		const cache = await caches.open(CACHE_NAME);
+		await cache.addAll(to_cache);
+	}
+
+	event.waitUntil(addFilesToCache());
+});
+
+self.addEventListener('activate', (event) => {
+	// Remove previous cached data from disk
+	async function deleteOldCaches() {
+		const keyList = await caches.keys();
+		const cachesToDelete = keyList.filter((key) => key !== CACHE_NAME);
+		await Promise.all(cachesToDelete.map((key) => caches.delete(key)));
+	}
+
+	event.waitUntil(deleteOldCaches());
+});
+
+self.addEventListener('fetch', (event) => {
+	// Try to get the response from the cache, add to cache if not found
+	async function addToCache(request, response) {
+		const cache = await caches.open(CACHE_NAME);
+		await cache.put(request, response);
+	}
+
+	async function cacheFirst(request) {
+		const responseFromCache = await caches.match(request);
+		if (responseFromCache) {
+			return responseFromCache;
+		}
+		const response = await fetch(request);
+		addToCache(request, response.clone());
+		return response;
+	}
+
+	event.respondWith(cacheFirst(event.request));
+});
+```
+
+> Careful with caching too much for too long! Browsers have a limit on the amount they cache, and not everything should be cached, for example requests that contain dynamic data that change over time
+
+Because service workers need to be bundled (since browsers don't yet support `import` in this context), **SvelteKit's service workers only work in the production build, not in development**, since it depends on the client-side app's build manifest. To test it locally, use `vite preview` after running a build.
+
+SvelteKit's service worker implementation is deliberately low-level. If you need a more full-flegded but also more opinionated solution, we recommend looking at solutions like [Vite PWA plugin](https://vite-pwa-org.netlify.app/frameworks/sveltekit.html), which uses [Workbox](https://web.dev/learn/pwa/workbox). For more general information on service workers, we recommend [the MDN web docs](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API/Using_Service_Workers).
```

Comment:
we could add a heading here to make it linkable from above

```suggestion
## External solutions

SvelteKit's service worker implementation is deliberately low-level. If you need a more full-flegded but also more opinionated solution, we recommend looking at solutions like [Vite PWA plugin](https://vite-pwa-org.netlify.app/frameworks/sveltekit.html), which uses [Workbox](https://web.dev/learn/pwa/workbox). For more general information on service workers, we recommend [the MDN web docs](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API/Using_Service_Workers).
```

Reply:
Given that the section isn't very big I'm not sure we need this. "On this page" would look rather empty, and the only entry being "External Solutions" feels kinda weird. So if we want to, we would need another heading somewhere.

Reply:
yeah, that's a fair point. other ideas for headings could be:
- creation
- registration (probably a sub-heading of the previous one)
- example

Reply:
I'll try to release the pwa plugin for kit next week. I need to merge a few PR and publish a New pwa plugin version before releasing pwa plugin integrations.
With https://github.com/vite-pwa/vite-plugin-pwa/pull/390 we can use something similar to kit sw, just check the sw in the PR example.